src/torchmetrics/utilities/compute.py

```diff
@@ -56,8 +56,8 @@ def _safe_divide(num: Tensor, denom: Tensor, zero_division: float = 0.0) -> Tens
     """
     num = num if num.is_floating_point() else num.float()
     denom = denom if denom.is_floating_point() else denom.float()
-    zero_division = torch.tensor(zero_division).float().to(num.device)
-    return torch.where(denom != 0, num / denom, zero_division)
+    zero_division_tensor = torch.tensor(zero_division, dtype=num.dtype).to(num.device, non_blocking=True)
```

Comment:
Renamed `zero_division` due to pylance complaining about re-typing the variable (as it was declared as `float` in the signature but is being assigned a `torch.Tensor`.

If so desired, alternatives include...
 - not rename it
 - retype the signature to Union[float, torch.Tensor]